lib/_http_outgoing.js

```diff
@@ -810,17 +810,18 @@ OutgoingMessage.prototype.end = function end(chunk, encoding, callback) {
     encoding = null;
   }
 
-  if (this.socket) {
-    this.socket.cork();
-  }
-
   if (chunk) {
     if (this.finished) {
       onError(this,
               new ERR_STREAM_WRITE_AFTER_END(),
               typeof callback !== 'function' ? nop : callback);
       return this;
     }
+
+    if (this.socket) {
+      this.socket.cork();
```

Comment:
Shouldn't **corking** the socket also include the last [case](https://github.com/nodejs/node/pull/36633/files#diff-48d21edbddb6e855d1ee5716c49bcdc0d913c11ee8a24a98ea7dbc60cd253556R836) of `switch` statement where the header is written when no chunk and request is not finished?